### PR TITLE
Pay-In API Frontend, Part One

### PR DESCRIPTION
**Pull request by zfox23**
Implements first technical subsection of the [Pay-In API scope doc](https://highfidelity.atlassian.net/wiki/spaces/HOME/pages/427524112/Pay-In+API).

*****

Open "Transfer HFC or Item Asset" Screen from Script:
```
var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
tablet.loadQMLSource("hifi/commerce/common/sendAsset/SendAsset.qml");
tablet.sendToQml({method: 'updateSendAssetQML',
	assetCertID: "<some cert ID; leave blank if sending HFC>"
    amount: "<amount of HFC to send; won't be editable by user. will always be "1" when gifting non-HFC assets>",
    username: "<some username; won't be editable by user>",
    message: "<some optional message; won't be editable by user>"
});
```

Example 1 (send 1 HFC to "zfoxStaging1" on the Staging metaverse with an explanitory message):
```
var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
tablet.loadQMLSource("hifi/commerce/common/sendAsset/SendAsset.qml");
tablet.sendToQml({method: 'updateSendAssetQML',
	assetCertID: "",
    amount: "1",
    username: "zfoxStaging1",
    message: "Pay me 1 HFC to start the roulette wheel!"
});
```

***********************************

Open "Purchase Asset from Marketplace" Screen from Script:
```
var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
tablet.loadQMLSource("hifi/commerce/checkout/Checkout.qml");
tablet.sendToQml({method: 'updateCheckoutQMLItemID', params: {itemId: "<some item ID>"}});
```

Example 1 (purchase "TTS v1.0" app from Staging Marketplace):
```
var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
tablet.loadQMLSource("hifi/commerce/checkout/Checkout.qml");
tablet.sendToQml({method: 'updateCheckoutQMLItemID', params: {itemId: "33687f5f-f85e-4e9c-ba19-8c7658038476"}});
```

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15400/HighFidelity-Beta-PR14476-a9fd443.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15400/HighFidelity-Beta-PR14476-a9fd443.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15400/HighFidelity-Beta-Interface-PR14476-a9fd443.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15400/HighFidelity-Beta-PR14476-a9fd443.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15400/HighFidelity-Beta-Interface-PR14476-a9fd443.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15437/HighFidelity-Beta-PR14476-67dfcca.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15437/HighFidelity-Beta-PR14476-67dfcca.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15437/HighFidelity-Beta-Interface-PR14476-67dfcca.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15437/HighFidelity-Beta-PR14476-67dfcca.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15437/HighFidelity-Beta-Interface-PR14476-67dfcca.exe

**Comment by rickcarr044** [comment]
Currently testing PR14476

File: interface/resources/qml/hifi/commerce/checkout/Checkout.qml [hunk 1]
```diff
@@ -1063,10 +1069,27 @@ Rectangle {
...
+
+    HifiCommon.RootHttpRequest {
+        id: http;
+    }
 
     //
     // FUNCTION DEFINITIONS START
     //
+
+    function updateCheckoutQMLFromHTTP(error, result) {
```
[the remark this answers is not in the document]
Comment (zfox23) [reply interface/resources/qml/hifi/commerce/checkout/Checkout.qml]: the QML would not be in a good state. thanks for the poke, i'll add some error handling

**Comment by zfox23** [comment]
@shanzzam and @rickcarr044 could you please re-approve? i added some error handling after @roxanneskelly's CR.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15458/HighFidelity-Beta-PR14476-a4175b0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15458/HighFidelity-Beta-Interface-PR14476-a4175b0.dmg

**Comment by zfox23** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15469/HighFidelity-Beta-PR14476-a4175b0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15469/HighFidelity-Beta-PR14476-a4175b0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15469/HighFidelity-Beta-Interface-PR14476-a4175b0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15469/HighFidelity-Beta-PR14476-a4175b0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15469/HighFidelity-Beta-Interface-PR14476-a4175b0.exe

**Comment by rickcarr044** [comment]
Currently testing PR 14476

**Comment by rickcarr044** [comment]
Note:  Error checking will need some work in subsequent PR(s)

For example - where Username was not included in the script, brings up the dialog:

sendmoneyto-script

Clicking on the Connections icon spins without bringing up anyone on the list.
sendmoneyto-spin

**Comment by zfox23** [comment]
thanks for finding that, Rick! i will make sure to add error handling if the script doesn't specify a username.

**Pull request merged**